packages/extension-vscode/src/server.ts

```diff
@@ -207,6 +207,42 @@ connection.onInitialize((params) => {
     // TODO: Support multiple workspaces (`params.workspaceFolders`).
     workspace = params.rootPath || '';
 
+    /**
+     * The `local` connector uses `jsdom` that is going to always try to load
+     * `canvas` even though it is not needed in the VS Code extension. If there
+     * is a mismatch between the user's node version and the one uses by
+     * VS Code, `canvas` will fail to load and crash the extension. To avoid
+     * that we hijack `require`'s cache and set an empty `Module` so `jsdom`
+     * doesn't use it and continues executing normally.
+     *
+     * Because `jsdom` is the one requireing `canvas` we search for `canvas`
+     * as if we were that package to make sure to find the same package.
+     *
+     */
+    const Module = require('module');
+
+    const jsdomPath: string = require.resolve('jsdom', { paths: [path.join(workspace, 'node_modules')] });
+    const jsdomNodeModules = jsdomPath.substring(0, jsdomPath.indexOf('jsdom'));
+
```

Comment:
👆 Does that snip the path before the jsdom folder?
```js
"/the/path/to/jsdom/index.js"
// substringed to
"/the/path/to/" // ?
```

Reply:
Yes. In this case it's `/the/path/to/node_modules/jsdom/` ➡️`/the/path/to/node_modules/`